snyk/cli
# feat: Applied the new IaC test summary formatter to the test flow  [CFG-1571]

### What does this PR do?
- Applies the new test IaC test summary formatted created in #3098 in the test flow of Snyk CLI, under the `iacCliOutput` feature flag, by:
    - Printing the new test summary section at the end of the test output, with a separator above it.
    - Removing the per-file meta section.
    - Removing the per-file summary message.

### Notes for the reviewer
- This formatter was created in a previous PR: #3098
- Following the discussion on [this Slack thread](https://snyk.slack.com/archives/C030UA3U1RB/p1649151568166039), the project name and project path were not added to the test summary as part of this ticket, but need to be added at a later point in time.
- For the `test/jest/acceptance/iac/test-terraform-var-deref.spec.ts`, a configuration was made in the fake server, which sets all feature flag values as `true`, including the `iacCliOutput` feature flag. This broke the tests of this file when introducing the new test summary, so the both the test file and the fake server were updated to follow our recent practices with feature flags.

### Where should the reviewer start?

- `src/cli/commands/test/index.ts`

### How should this be manually tested?

1. Make sure the `iacCliOutput` feature flag is enabled for your org.
2. In the Snyk CLI, execute an IaC test with the following command:
    ```
    snyk-dev iac test <file-or-dir>
    ```
3. Verify that:
    - The meta section after each project is not displayed.
    - The summary sentence after each project is not displayed.
    - The new test summary section at the end of the output is displayed with:
        -  All the valid values.
        - The expected design.
4. Disable the `iacCliOutput` feature flag for your org.
5. In the Snyk CLI, execute an IaC test with the following command:
    ```
    snyk-dev iac test <file-or-dir>
    ```
3. Verify that:
    - The meta section after each project is displayed.
    - The summary sentence after each project is displayed.
    - The new test summary section at the end of the output is not displayed.

### Any background context you want to provide?

As part of the MVP for the new CLI output, we’d like to replace the current per-project summaries in the current IaC test output with a single aggregated summary at the end of the test output. 
This can be viewed in the following design, taken from [the MVP definition](https://miro.com/app/board/uXjVOOpVGxU=/?moveToWidget=3458764518430874390&cot=14): 

![image](https://user-images.githubusercontent.com/46415136/162007885-4fc89eaf-bd58-46fe-aae9-12fdc0a3ba75.png)

### What are the relevant tickets?

- [CFG-1571](https://snyksec.atlassian.net/browse/CFG-1571)

### Additional information

- [Slack thread](https://snyk.slack.com/archives/C030UA3U1RB/p1649151568166039)
- A preceding PR: #3098 
- [Figma design](https://www.figma.com/file/buNFZqr1Uhf4SaifHw4WXE/CLI-IaC-Output-Improvements)
- 
<details>
<summary><h2>Screenshots</h2></summary>

- With a valid file with issues:

    ![image](https://user-images.githubusercontent.com/46415136/162258356-506ac849-78e0-4a8b-9cd4-7e8e4abc166e.png)

- With a valid file without issues:

    ![image](https://user-images.githubusercontent.com/46415136/162258555-ad217c69-239e-445c-bf99-9db3810d5289.png)

- With an invalid file:

     ![image](https://user-images.githubusercontent.com/46415136/162259010-caea13e2-602a-48d9-bf13-faede88343af.png)

- With a directory with only valid files:

    ![image](https://user-images.githubusercontent.com/46415136/162018049-28b32f3d-000f-4a4f-901d-054d36f76155.png)

- With a valid directory without issues:

    ![image](https://user-images.githubusercontent.com/46415136/162259156-5cf4ef05-ee11-45c6-b4ff-aecf371f5afc.png)

- With a directory with valid and invalid files:

    ![image](https://user-images.githubusercontent.com/46415136/162259935-e8522542-5b0d-41a3-85d5-e29a50ddfc0f.png)
</details>
